mllib/src/main/scala/org/apache/spark/mllib/clustering/LocalKMeans.scala

```diff
@@ -46,7 +47,9 @@ private[mllib] object LocalKMeans extends Logging {
 
     // Initialize centers by sampling using the k-means++ procedure.
     centers(0) = pickWeighted(rand, points, weights).toDense
-    val costArray = points.map(EuclideanDistanceMeasure.fastSquaredDistance(_, centers(0)))
+    val pointsParVector = new ParVector(points.toVector)
```

Comment:
Can you find the threshold for `points.length` that can get performance improvement?

Not sure this always can get performance improvement, `points.toVector` will also be some costs,  it will create a new Vector and add all elements using tail recursion.

Reply:
Thanks for reply, I think the benefits of parallelism outweigh the costs of vector copying.
It's not easy to find the threshold, because the points are two-dimensionsal.